org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -3164,15 +3202,37 @@ protected void saveNew_afterSetID()
 	{
 		
 	}
+	
+	/**
+	 * 	Create Single/Multi Key Where Clause
+	 * 	@param withValues if true uses actual values otherwise ?
+	 * 	@return where clause
+	 */
+	public String get_WhereClause (boolean withValues) {
+		return get_WhereClause(withValues,null);
+	}
 
 	/**
 	 * 	Create Single/Multi Key Where Clause
 	 * 	@param withValues if true uses actual values otherwise ?
+	 *  @param uuID RecordUU
 	 * 	@return where clause
 	 */
-	public String get_WhereClause (boolean withValues)
+	public String get_WhereClause (boolean withValues, String uuID)
 	{
 		StringBuilder sb = new StringBuilder();
+
+		if (!Util.isEmpty(uuID, true))
+		{
+			sb.append(getUUIDColumnName()).append("=");
+			if (withValues)
+				sb.append(uuID);
```

Comment:
this is not correct - you need to  quote the uuID string value, i.e DB.TO_STRING(uuID)

Reply:
Done